tests/units/cli/test_commands.py

```diff
@@ -662,3 +678,202 @@ def test_run_command(self):
+            invalid_json_file = f'{temp_dir}/invalid_file.json'
+            valid_json_file = f'{temp_dir}/valid_file.json'
+            empty_file = f'{temp_dir}/empty_stat.json'
+
+            with open(valid_json_file, 'w', encoding='utf-8') as valid_file:
+                json.dump({'foo': 'bar'}, valid_file)
+
+            # 1. asserting if raises exception on invalid contents
+            for invalid_content in invalid_file_contents:
+                with open(invalid_json_file, 'w', encoding='utf-8') as invalid_file:
+                    invalid_file.write(invalid_content)
+
+                with pytest.raises(commands.RunCommandException) as command_exception:
+                    self._assert_tap_config(
+                        config=valid_json_file,
+                        properties=valid_json_file,
+                        state=invalid_json_file,
+                    )
+                assert str(command_exception.value) == f'Invalid json file for state: {invalid_json_file}'
+
+            # 2. asserting if it is valid with empty file
+            with open(empty_file, 'w', encoding='utf-8') as state_file:
+                state_file.write('')
+            self._assert_tap_config(
+                config=valid_json_file,
+                properties=valid_json_file,
+                state=empty_file
+            )
+
+    def test_tap_config_valid_if_json_property_is_none(self):
+        """Test TapConfig is valid if a json property is None"""
+        with TemporaryDirectory() as temp_dir:
+            valid_json_file = f'{temp_dir}/valid_file.json'
+            with open(valid_json_file, 'w', encoding='utf-8') as valid_file:
+                json.dump({'foo': 'bar'}, valid_file)
+
+            test_cases = (
+                (None, valid_json_file, valid_json_file),
+                (valid_json_file, None, valid_json_file),
+                (valid_json_file, valid_json_file, None)
+            )
+            for config, properties, state in test_cases:
+                self._assert_tap_config(
+                    config=config,
+                    properties=properties,
+                    state=state
+                )
+
+    def test_tap_config_raise_exception_if_not_valid_json_after_retries(self):
+        """Test it raises and exception if invalid json files yet after many retries"""
+        invalid_file_contents = ['', ' ', 'foo', '{"foo": 1']
+
+        with TemporaryDirectory() as temp_dir:
+            invalid_json_file = f'{temp_dir}/invalid_file.json'
+            valid_json_file = f'{temp_dir}/valid_file.json'
+
+            with open(valid_json_file, 'w', encoding='utf-8') as valid_file:
+                json.dump({'foo': 'bar'}, valid_file)
+
+            for invalid_content in invalid_file_contents:
+                with open(invalid_json_file, 'w', encoding='utf-8') as invalid_file:
+                    invalid_file.write(invalid_content)
+
+                for case_number in range(2):
```

Comment:
same old comment about cases and for loops

Reply:
- [ ] Fix it